src/bigint.ts

```diff
@@ -0,0 +1,30 @@
+import type { Scalar, ScalarTag } from 'yaml'
+import { StringifyContext, stringifyString } from 'yaml/util'
+
+/**
+ * `!bigint` BigInt
+ *
+ * [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) values,
+ * using their conventional `123n` representation.
+ */
+export const bigint: ScalarTag = {
+  identify: (value: any) => {
+    return typeof value === 'bigint' || value instanceof BigInt
+  },
+  tag: '!bigint',
+  resolve(str: string) {
+    const match = str.match(
+      /^(-?)([0-9]*|0x[0-9a-fA-F]+|0o[0-7]+|0b[01]+)n?$/
```

Comment:
I think this matches just `-` and an empty string, which isn't intentional.

I'm also not convinced about supporting negatives for non-decimals, as e.g. `BigInt('-0x1')` throws an error. While YAML numbers support negative non-decimal values, this encoding is specifically for JS BigInt, for which there's an [exact spec](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-stringtobigint) for valid input. We should not extend support for parsing other strings outside that.

Reply:
Ah, I didn't realize you couldn't do negative hex/oct/bin numbers, since that works for non-bigint.

Seems like this matches it?

```patch
diff --git a/src/bigint.ts b/src/bigint.ts
index 8444f72..955281a 100644
--- a/src/bigint.ts
+++ b/src/bigint.ts
@@ -14,11 +14,10 @@ export const bigint: ScalarTag = {
   tag: '!bigint',
   resolve(str: string) {
     const match = str.match(
-      /^(-?)([0-9]*|0x[0-9a-fA-F]+|0o[0-7]+|0b[01]+)n?$/
+      /^(-?[0-9]+|0x[0-9a-fA-F]+|0o[0-7]+|0b[01]+)n?$/
     )
     if (!match) throw new Error('Invalid BigInt value')
-    const bi = BigInt(match[2])
-    return match[1] ? bi * -1n : bi
+    return BigInt(match[1])
   },
   stringify(item: Scalar, ctx: StringifyContext, onComment, onChompKeep) {
     if (!bigint.identify?.(item.value)) {
```

Or I suppose it could just accept anything and throw if the cast fails, which would avoid the need to even have a regexp?

Reply:
Oh, you _can_ do negative bin/oct/hex bigints as literals, just not with the string cast.  Ie:

```
> -0xDEADBEEFn
-3735928559n
> BigInt('-0xDEADBEEF')
Uncaught SyntaxError: Cannot convert -0xDEADBEEF to a BigInt
```

So I guess the question is whether the restriction on the yaml expression should be "code that is a bigint literal" or "a string that can be passed to the BigInt function".

If it's the first, then `-0xf` should be allowed. Since it includes the trailing `n` by default in the stringified form, it seemed like that would be the expectation.

Reply:
Oh, but, `yaml.parse('-0xf')` parses as a string.  So yeah, probably better to not support it for BigInt if it's not supported for Number.

Reply:
> Or I suppose it could just accept anything and throw if the cast fails, which would avoid the need to even have a regexp?

I like that idea. Could even drop the try-catch, as the errors thrown by `BigInt()` look pretty decent.

I'm a bit surprised that `BigInt('')` parses to `0n`, happy to error on that too.